src/core/thread/network_data_leader.cpp

```diff
@@ -546,11 +454,10 @@ void Leader::RemoveBorderRouter(uint16_t aRloc16)
         {
             mStableVersion++;
         }
-
-        ConfigureAddresses();
     }
 
     mMle.HandleNetworkDataUpdate();
+    mNetif.SetStateChangedFlags(OT_THREAD_NETDATA_UPDATED);
```

Comment:
How about add one more condition`if (rlocIn)` for Line 460?
RemoveBorderRouter() would be called when Leader ReleaseRouterId(), however it is possible the releasing router id is not a border router, which should not be counted as network data update.